saleor/graphql/account/mutations/permission_group/permission_group_create.py

```diff
@@ -213,54 +212,32 @@ def clean_channels(
         """Clean adding channels when the group hasn't restricted access to channels."""
         user = info.context.user
         user = cast(User, user)
-        if cleaned_input.get("restricted_access_to_channels") is False:
-            if not user.is_superuser:
-                channel_ids = set(Channel.objects.values_list("id", flat=True))
-                accessible_channel_ids = {
-                    channel.id for channel in user_accessible_channels
-                }
-                not_accessible_channels = set(channel_ids - accessible_channel_ids)
-                error_code = PermissionGroupErrorCode.OUT_OF_SCOPE_CHANNEL.value
-                if not_accessible_channels:
-                    raise ValidationError(
-                        {
-                            "restricted_access_to_channels": ValidationError(
-                                "You can't manage group with channels out of "
-                                "your scope.",
-                                code=error_code,
-                            )
-                        }
-                    )
-            cleaned_input["add_channels"] = []
-        elif add_channels := cleaned_input.get("add_channels"):
+
+        if "restricted_access_to_channels" in cleaned_input:
             cls.ensure_can_manage_channels(
-                user, user_accessible_channels, errors, add_channels
+                user, errors, "restricted_access_to_channels"
             )
+            if cleaned_input.get("restricted_access_to_channels") is False:
+                cleaned_input["add_channels"] = []
+
+        elif cleaned_input.get("add_channels"):
+            cls.ensure_can_manage_channels(user, errors, "add_channels")
 
     @classmethod
-    def ensure_can_manage_channels(
-        cls,
-        user: "User",
-        user_accessible_channels: List["Channel"],
-        errors: dict,
-        channels: List["Channel"],
-    ):
-        # user must have access to all channels from `add_channels` list
+    def ensure_can_manage_channels(cls, user: "User", errors: dict, field: str):
         if user.is_superuser:
             return
-        channel_ids = {str(channel.id) for channel in channels}
-        accessible_channel_ids = {
-            str(channel.id) for channel in user_accessible_channels
-        }
-        invalid_channel_ids = channel_ids - accessible_channel_ids
-        if invalid_channel_ids:
-            ids = [
-                graphene.Node.to_global_id("Channel", pk) for pk in invalid_channel_ids
-            ]
-            error_msg = "You can't add channel that you don't have access to."
-            code = PermissionGroupErrorCode.OUT_OF_SCOPE_CHANNEL.value
-            params = {"channels": ids}
-            cls.update_errors(errors, error_msg, "add_channels", code, params)
+
+        if user.has_perm(ChannelPermissions.MANAGE_CHANNELS):
+            return
+
```

Comment:
Why do you remove the whole check if the user can access the given channel? Was the change of assumption discussed somewhere?